### PR TITLE
Fixes broken CPUParticles2D AtlasTextures usage

### PR DESCRIPTION
**Pull request by bitstopper**

### CODE REVIEW

**Comment by akien-mga** [comment]
As mentioned in #32459, this should likely be implemented for `Particles2D` too.

Note that you added a `tools=yes` file by mistake, the commit should be amended to remove it.

And see comments on #32351, change receptors are editor-only and should not be used for runtime code. (#32351 would now even disable them.)

**Comment by bitstopper** [comment]
Particles2D and CPUParticles2D act both wrong on AtlasTextures at the moment but everyone in their own way. Particles2D shows the whole Atlas and doesn't take size into consideration and CPUParticles2D did respect the size but scaled the whole atlas down.

The bug report for Particles2D & AtlasTextures: https://github.com/godotengine/godot/issues/13923

File: scene/2d/cpu_particles_2d.cpp [hunk 1]
```diff
@@ -33,6 +33,7 @@
 #include "scene/2d/canvas_item.h"
 #include "scene/2d/particles_2d.h"
 #include "scene/resources/particles_material.h"
+#include "core/core_string_names.h"
```
Comment (akien-mga) [inline scene/2d/cpu_particles_2d.cpp]: Includes should be ordered alphabetically, so this failed `clang-format` style checks.

**Review by clayjohn — APPROVED** [review]
Changes look good. But Particles2D also need a similar fix. We can open a bug report once this is merged.

**Comment by akien-mga** [comment]
Well, no luck for GPU Particles so far so I rest my case. A partial fix is better than no fix indeed :)

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.1.

**Pull request merged**